src/units.rs

```diff
@@ -377,6 +378,36 @@ pub fn to_ideal_unit(number: Number) -> Number {
     } else {
       return Number::new(value, Millimeter)
     }
+  } else if number.unit.category() == Time {
+    if value >= d128!(31556952000000000000) {
+      return Number::new(value/Millienium.weight(), Millenium);
+    } else if value >= d128!(3155695200000000000) {
+      return Number::new(value/Century.weight(), Century);
+    } else if value >= d128!(315569520000000000) {
+      return Number::new(value/Decade.weight(), Decade);
+    } else if value >= d128!(31556952000000000) {
+      return Number::new(value/Year.weight(), Year);
+    } else if value >= d128!(7889238000000000) {
+      return Number::new(value/Quarter.weight(), Quarter);
+    } else if value >= d128!(2629746000000000) {
+      return Number::new(value/Month.weight(), Month);
+    } else if value >= d128!(604800000000000) {
+      return Number::new(value/Week.weight(), Week);
+    } else if value >= d128!(86400000000000) {
```

Comment:
I think it's better if we don't return Millenium, Century, Decade, Quarter, Month or Week. In many cases, these might not fit, so Year and Day should be safe picks.

(Also, typo: `Millienium` should be `Millenium`. And I just realied that `Millenium` is a typo too, but that's my fault)

Reply:
I'm not sure how these compile errors didn't get picked up before I submitted the patch :S I'm sorry